src/main/java/prefs/Preferences.java

```diff
@@ -11,122 +17,243 @@
+     * @param sessionConfigFileName The name of the session config file
+     * @param userConfigFileName The name of the user config file
+     * @param ignoreExisting True if existing config file is to be ignored.
+     */
+    private Preferences(String configDirectory,
+                        String sessionConfigFileName,
+                        String userConfigFileName,
+                        boolean ignoreExisting) {
+        this.configDirectory = configDirectory;
+        this.sessionConfigFileName = sessionConfigFileName;
+        this.userConfigFileName = userConfigFileName;
+
+        if (ignoreExisting) {
+            this.sessionConfig = createConfig(DEFAULT_FILE_CONTENTS, SessionConfig.class);
+            this.userConfig = createConfig(DEFAULT_FILE_CONTENTS, UserConfig.class);
         } else {
-            loadGlobalConfig();
+            this.sessionConfig = loadConfig(configDirectory, sessionConfigFileName, SessionConfig.class);
+            this.userConfig = loadConfig(configDirectory, userConfigFileName, UserConfig.class);
         }
     }
 
     /**
-     * Initialises a Preferences instance which creates its config file, or loads
-     * from it if it already exists.
+     * Creates a new Config instance
+     * @param contents The contents of the object to create
+     * @param configClass The class of the Config instance to create
+     * @return The created Config instance
      */
-    public static Preferences load(String configFileName) {
-        return new Preferences(configFileName, false);
+    private <T> T createConfig(String contents, Class<T> configClass) {
+        return new JsonHelper().fromJsonString(contents, configClass);
     }
 
     /**
-     * Initialises a Preferences instance which always creates its config file.
-     * This will overwrite it with a default configuration if it already exists.
+     * Loads a Config instance from a file on disk
+     * @param configDirectory The directory that the config file is held in.
+     * @param configFileName The filename of the config file on disk
+     * @param configClass The class of the config to load
+     * @return The loaded Config instance
      */
-    public static Preferences create(String configFileName) {
-        return new Preferences(configFileName, true);
-    }
+    private <T> T loadConfig(String configDirectory, String configFileName, Class<T> configClass) {
+        FileHelper configFileHelper = new FileHelper(configDirectory, configFileName);
+        String fileContents = DEFAULT_FILE_CONTENTS;
+        if (configFileHelper.exists()) {
+            try {
+                fileContents = configFileHelper.loadFileContents();
+            } catch (IOException e) {
+                // if we can't read the file, just ignore the existing file
+                HTLog.error(logger, e);
+                logger.error(e.toString());
+                fileContents = DEFAULT_FILE_CONTENTS;
+            }
```

Comment:
I feel that it should be done in another PR, as it is a little out of scope. The old config file handler also swallowed exceptions. A new issue can be made can be to rectify this. Won't this be a better idea?